docs/nnx/nnx_basics.md

```diff
@@ -252,16 +254,86 @@ print(f'{y.shape = }')
 nnx.display(model)
 ```
 
-How do NNX transforms achieve this? To understand how NNX objects interact with
-JAX transforms lets take a look at the Functional API.
+## Using Modules as Pytrees
 
-+++
+Modules are not pytrees by design, however when needed `nnx.Pytree` can be used to obtain a pytree 
+representation of a Module. Pytree forwards all attribute access to the underlying Module so you can
+naturally use it as if it were the original Module.
+
+```{code-cell} ipython3
+model = nnx.Pytree(nnx.Linear(3, 3, rngs=nnx.Rngs(0)))
+model = jax.tree.map(lambda x: x, model) # its a pytree!
+
+# attribute access
+print(f'{model.kernel.shape = }')
+nnx.display(model)
+```
+
+Pytree instances cannot share references with other objects, their state is not automatically
+propagated by NNX transforms, and they do not support mutation. If mutation is needed the Pytree
+can be used as a context manager to get a reference to the underlying object:
+
+```{code-cell} ipython3
+model = nnx.Pytree(nnx.Linear(3, 3, rngs=nnx.Rngs(0)))
+
+try:
+  model.in_features = 10
+except AttributeError as e:
+  print('AttributeError:', e)
+
+with model as m:
+  m.in_features = 10
+
+print(f'{model.in_features = }')
+```
+
+All update will be propagated to the Pytree at the end of the context.
+
+### Integration Example
+
+Finally lets see an example of how to use Pytree to use a Module with the `odeint` function,
+a common use case when sampling from Diffusion Models.
+
+```{code-cell} ipython3
+from jax.experimental.ode import odeint
+
+model = nnx.Pytree(nnx.Linear(3, 3, rngs=nnx.Rngs(0)))
+
+def step_fn(y, t, model):
+  return model(y) # dy/dt
+
+y = jnp.ones((3,))
+ts = jnp.linspace(0, 1, 5)
+ts = jnp.linspace(0, 1, 5)
+
+ys = odeint(step_fn, y, ts, model)
+
+print(ys)
+```
+
```

Comment:
Add a line to introduce this piece of code, something like: Let's see another example mixing NNX and Equinox modules, as Equinox modules are pure pytrees.